hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -70,120 +63,61 @@
 			OptionValidator validator,
 			TransactionContext txnCtx,
+			TransactionRecordService transactionRecordService,
 			GlobalDynamicProperties dynamicProperties
 	) {
-		this.ledger = ledger;
+		this.accountStore = accountStore;
 		this.validator = validator;
 		this.txnCtx = txnCtx;
+		this.transactionRecordService = transactionRecordService;
 		this.dynamicProperties = dynamicProperties;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			final var op = txnCtx.accessor().getTxn().getCryptoUpdateAccount();
-			final var target = op.getAccountIDToUpdate();
-			final var customizer = asCustomizer(op);
-
-			if (op.hasExpirationTime() && !validator.isValidExpiry(op.getExpirationTime())) {
-				txnCtx.setStatus(INVALID_EXPIRATION_TIME);
-				return;
-			}
-			final var validity = sanityCheck(target, customizer);
-			if (validity != OK) {
-				txnCtx.setStatus(validity);
-				return;
-			}
+		/* --- Translate from gRPC types --- */
+		final var txnBody = txnCtx.accessor().getTxn();
+		final var op = txnBody.getCryptoUpdateAccount();
 
-			ledger.customize(target, customizer);
-			txnCtx.setStatus(SUCCESS);
-		} catch (MissingAccountException mae) {
-			txnCtx.setStatus(INVALID_ACCOUNT_ID);
-		} catch (DeletedAccountException aide) {
-			txnCtx.setStatus(ACCOUNT_DELETED);
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
-		}
-	}
-
-	private ResponseCodeEnum sanityCheck(AccountID target, HederaAccountCustomizer customizer) {
-		if (!ledger.exists(target) || ledger.isSmartContract(target)) {
-			return INVALID_ACCOUNT_ID;
-		}
+		/* --- Load the model objects --- */
+		final var accountId = Id.fromGrpcAccount(op.getAccountIDToUpdate());
+		final var accountToBeUpdated = accountStore.loadAccount(accountId);
 
-		final var changes = customizer.getChanges();
-		final var keyChanges = customizer.getChanges().keySet();
+		validateFalse(accountToBeUpdated.isSmartContract(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 
-		if (ledger.isDetached(target)) {
-			if (!keyChanges.equals(EXPIRY_ONLY)) {
-				return ACCOUNT_EXPIRED_AND_PENDING_REMOVAL;
-			}
+		if (op.hasExpirationTime()) {
+			validateFalse(accountToBeUpdated.getExpiry() > op.getExpirationTime().getSeconds(), EXPIRATION_REDUCTION_NOT_ALLOWED);
+			validateFalse(!validator.isValidExpiry(op.getExpirationTime()), INVALID_EXPIRATION_TIME);
 		}
 
-		if (keyChanges.contains(AccountProperty.EXPIRY)) {
-			final long newExpiry = (long)changes.get(AccountProperty.EXPIRY);
-			if (newExpiry < ledger.expiry(target)) {
-				return EXPIRATION_REDUCTION_NOT_ALLOWED;
-			}
+		if (op.hasMaxAutomaticTokenAssociations()) {
+			validateFalse(op.getMaxAutomaticTokenAssociations().getValue() < accountToBeUpdated.getAlreadyUsedAutomaticAssociations(), EXISTING_AUTOMATIC_ASSOCIATIONS_EXCEED_GIVEN_LIMIT);
+			//TODO dynamicProperties.maxTokensPerAccount() returns inconsistent values
```

Comment:
Is this a leftover?

Reply:
there was an issue with an e2e test case, it is now extracted into a separate suite